xarray/core/dataset.py

```diff
@@ -3150,11 +3150,25 @@ def drop(self, labels, dim=None, *, errors='raise'):
             any of the variable or index labels passed are not
             in the dataset. If 'ignore', any given labels that are in the
             dataset are dropped and no error is raised.
+        **dim_kwargs : {dim: label, ...}, optional
+            The keyword arguments form of ``dim`` and ``labels``.
+            Either ``dim`` and ``labels`` or ``dim_kwargs`` must be provided.
 
         Returns
         -------
         dropped : Dataset
         """
+        if len(dim_kwargs) > 0:
+            if labels is not None or dim is not None:
+                raise ValueError('cannot specify both `dim` and `labels` along'
+                                 'with keyword arguments')
+            if len(dim_kwargs) > 1:
+                raise ValueError('cannot specify multiple dimensions')
+            dim = list(dim_kwargs.keys())[0]
+            labels = list(dim_kwargs.values())
+            if utils.is_list_like(labels[0]):
```

Comment:
I don't think we need this check -- we already handle both scalars and lists in the logic below.

Reply:
Since we convert `dim_kwargs`'s values to a list, if the user passes in a list, we get a list with a single element which is also a list. The logic below won't catch this list-in-list scenario. Happy to change this if you know a simpler way to handle `dict_values`. My problem was

```python
>>> x = {'foo': 1}
>>> x.values()
dict_values([1])
>>> x.values()[0] # TypeError
```

which is why I do `list(x.values())` instead.

Reply:
Ah, yes. I think the cleanest way to do this would be with tuple unpacking, e.g.,
```
((dim, labels),) = dim_kwargs.items()
```
or
```
(dim, labels) = list(dim_kwargs.items())[0]
```

Reply:
That works great. Thanks.